app/controllers/Admin.scala

```diff
@@ -207,5 +212,87 @@ object Admin extends Controller with PanDomainAuthActions {
     )
   }
 
+  /*
+    Section Tag association
+   */
+
+  val addSectionTagForm = Form(
+    mapping(
+      "section_id" -> longNumber,
+      "tag_id" -> nonEmptyText
+    )(assignTagToSectionFormData.apply)(assignTagToSectionFormData.unapply)
+  )
+
+  val removeSectionTagForm = Form(
+    mapping(
+      "section_id" -> longNumber,
+      "tag_id" -> nonEmptyText
+    )(unAssignTagToSectionFormData.apply)(unAssignTagToSectionFormData.unapply)
+  )
+
+  def sectionsAndTags(selectedSectionIdOption: Option[Long]) = (AuthAction andThen WhiteListAuthFilter).async {
+
+    val selectedSectionOptionFt:Future[Option[Section]] = selectedSectionIdOption match {
+      case Some(selectedSectionId) => {
+        for {
+          sections <- getSortedSections()
+        }yield{
+          Some(sections.filter( x => x.id == selectedSectionId ).head)
+        }
+      }
+      case None => Future(None)
+    }
+    val tagIdsOptionFuture: Future[Option[List[String]]] = selectedSectionIdOption match {
```

Comment:
Hi Luke. This is a good point. The intended semantics is that if the `selectedSectionId` was set (this is an optional value), then the `tagIds` are also going to be set (and otherwise are optional). So, originally, I made sense to me to have both `Some(thing)` at the same time, or both `None` at the same time. Also, later in the code the HTML code checks whether the list of tags is something or not before displaying a `<h3>Existing Tags</h3>` and then the list of tags. 

That having been said, your suggestion makes me thing that the list of tags should always be a list and the view's details driven by the value of `selectedSectionId`. 

I shall refactor this now :)